multiversx_sdk/multisig/resources.py

```diff
@@ -0,0 +1,231 @@
+        function_call = [function_name, *function_arguments]
+        return cls(to, tokens, function_call, gas_limit)
+
+
+class ProposeAsyncCallInput:
+    def __init__(
+        self,
+        to: Address,
+        function_call: list[bytes],
+        opt_gas_limit: Optional[int] = None,
+    ) -> None:
+        self.to = to
+        self.function_call = function_call
+        self.opt_gas_limit = opt_gas_limit
+
+    @classmethod
+    def new_for_transfer(
+        cls,
+        to: Address,
+        token_transfers: list[TokenTransfer],
+        gas_limit: Optional[int] = None,
+    ):
+        # Since multisig requires the transfer to be encoded as variadic<bytes> in "function_call",
+        # we leverage the transactions factory to achieve this (followed by splitting the data).
+        transactions_factory = TransferTransactionsFactory(TransactionsFactoryConfig(""))
+        transaction = transactions_factory.create_transaction_for_transfer(
+            sender=Address.empty(),
+            receiver=Address.empty(),
+            # Multisig wasn't designed to work with EGLD within MultiESDTNFT.
+            native_amount=0,
+            token_transfers=token_transfers,
+        )
+
+        function_call_parts = transaction.data.split(ARGS_SEPARATOR.encode())
+        function_name = function_call_parts[0]
+        function_arguments = [bytes.fromhex(item.decode()) for item in function_call_parts[1:]]
+        function_call = [function_name, *function_arguments]
+        return cls(to, function_call, gas_limit)
+
+    @classmethod
+    def new_for_transfer_execute(
+        cls,
+        to: Address,
+        token_transfers: list[TokenTransfer],
+        function: str,
+        arguments: list[Any],
+        gas_limit: Optional[int] = None,
+        abi: Optional[Abi] = None,
+    ):
+        # Since multisig requires the transfer & execute to be encoded as variadic<bytes> in "function_call",
+        # we leverage the transactions factory to achieve this (followed by splitting the data).
+        transactions_factory = SmartContractTransactionsFactory(TransactionsFactoryConfig(""), abi=abi)
+        transaction = transactions_factory.create_transaction_for_execute(
+            sender=Address.empty(),
+            contract=Address.empty(),
+            function=function,
+            gas_limit=0,
+            arguments=arguments,
+            native_transfer_amount=0,
+            token_transfers=token_transfers,
+        )
+
+        function_call_parts = transaction.data.split(ARGS_SEPARATOR.encode())
+        function_name = function_call_parts[0]
+        function_arguments = [bytes.fromhex(item.decode()) for item in function_call_parts[1:]]
+        function_call = [function_name, *function_arguments]
+        return cls(to, function_call, gas_limit)
+
+
+class ProposeSyncCallInput(ProposeAsyncCallInput):
+    pass
+
+
+class EsdtTokenPayment(StructValue):
```

Comment:
Does it work if we use a regular Python class (not a _StructValue_)?

Reply:
it does not if the abi is not provided